### PR TITLE
Upgrade Gradle to 5.0

### PR DESCRIPTION
**Pull request by swegner**
This is a major release upgrade, with many new features. See release
notes here: https://docs.gradle.org/5.0/release-notes.html

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by adude3141** [comment]
IIRC upgrade to gradle5.0 was blocked mainly by Gogradle plugin, which is not yet compatible to v5.0. Although work seems going on there: https://github.com/gogradle/gogradle/pull/270

**Comment by swegner** [comment]
@adude3141 thanks, good to know. 

I wanted to kick this off just to see how close we are. I don't think I'll have time to fix failures soon, so feel free to pick this up if you're interested.

File: build.gradle [hunk 1]
```diff
@@ -88,7 +88,9 @@ if (!gradle.startParameter.isOffline()) {
 //   dot -Tsvg build/reports/visteg.dot > build_dependencies.svg
 //
 // See https://github.com/mmalohlava/gradle-visteg for further details.
-apply plugin: "cz.malohlava.visteg"
```
Comment (swegner) [inline build.gradle]: This might be an alternative for visualizing task dependencies: https://github.com/dorongold/gradle-task-tree

**Comment by adude3141** [comment]
After looking back into https://issues.apache.org/jira/browse/BEAM-4042 there have been further problems with

- apt plugin
- findbugs plugin

apt could probably just be removed - not sure about intended eclipse support here, but otherwise annotation processing is natively supported by Gradle since a while

findbugs: not sure about status here. iirc that should be replaced by errorprone, we might need to push here a bit

**Comment by adude3141** [comment]
After looking into this again, current status:
- few minor issues of incompability - will prepare some PRs
- needs upgrade of used gradle plugins - will prepare PRs

Blockers:
- visteg plugin - this seems to be abandoned, but is incompatible to gradle5, we need to decide if we replace as suggested by @swegner, or simply drop
- Gogradle - was updated but we hit https://github.com/gogradle/gogradle/issues/276

Besides that gradle 5.1.1 seems to be building beam.

**Comment by swegner** [comment]
> visteg plugin - this seems to be abandoned, but is incompatible to gradle5, we need to decide if we replace as suggested by @swegner, or simply drop

I find it valuable to be able to visualize the task dependency tree when debugging build issues-- I've documented it in the wiki [here](https://cwiki.apache.org/confluence/display/BEAM/Gradle+Tips#GradleTips-VisualizingTaskDependencies).

If this is one of the last blockers to upgrading, let me see if I can migrate to an alternative plugin. I can work on it this week.

**Comment by adude3141** [comment]
Tried to update gradle-offline-dependencies-plugin to latest 0.5.0 version,
but hit https://github.com/mdietrichstein/gradle-offline-dependencies-plugin/issues/21

Confirmed that it would work with 5.1.1, so we have to delay that upgrade until we do the switch

**Comment by swegner** [comment]
Perhaps we should have multiple commits in a single PR do the full upgrade the Gradle HEAD:

1. Upgrade to Gradle 5.0, disable `gradle-offline-dependencies-plugin`
2. Upgrade to Gradle 5.1
3. Upgrade to Gradle 5.1.1, re-enable `gradle-offline-dependencies-plugin`

**Comment by swegner** [comment]
I'm going to close this PR for now as I don't currently have bandwidth to drive the upgrade. @adude3141 feel free to pick it up if you have time.

**Pull request closed without merging**